libscript/src/objc.lcb

```diff
@@ -610,4 +610,208 @@ public handler CreateObjcInformalDelegateWithContext(in pProtocol as List, in pH
+			into mNSConcreteGlobalBlock
+		CloseLibrary(tSystem)
+	end if
+end handler
+
+private constant kBlockSize is 32
+private constant kBlockDescriptorSize is 40
+
+public foreign type _ObjcBlockDescriptor binds to "MCAggregateTypeInfo:ffrrr"
+public foreign type _ObjcBlock binds to "MCAggregateTypeInfo:rEErr"
+public type ObjcBlockPointer is Pointer
+
+foreign handler MCHandlerGetFunctionPtr(in pHandler as any, out rPtr as Pointer) \
+	returns CBool \
+	binds to "<builtin>"
+
+private foreign handler CopyBlockDescriptor(inout pBlock as _ObjcBlockDescriptor, in pSize as UIntSize, out rNewBlock as Pointer) \
+	returns CBool \
+	binds to "MCMemoryAllocateCopy"
+
+private foreign handler CopyBlock(inout pBlock as _ObjcBlock, in pSize as UIntSize, out rNewBlock as Pointer) \
+	returns CBool \
+	binds to "MCMemoryAllocateCopy"
+
+private foreign handler MCMemoryDelete(in pBlock as Pointer) \
+	returns nothing \
+	binds to "<builtin>"
+
+private foreign handler BlockPointerToBlock(out rBlock as _ObjcBlock, in pPointer as ObjcBlockPointer, in pSize as UIntSize) \
+	returns nothing \
+	binds to "c:memcpy"
+
+/**
+Create an Objective-C block pointer that wraps an LCB handler.
+
+Example:
+private variable sRequestPermissionsCompletionHandler as optional ObjcBlockPointer
+private variable sTarget as ScriptObject
+
+public handler AudioLibraryInitialize() returns Boolean
+	if not CreateObjcBlockPointerFromHandler(RequestPermissionsCompletionHandler, sRequestPermissionsCompletionHandler) then
+		put nothing into sRequestPermissionsCompletionHandler
+		return false
+	end if
+	put the caller into sTarget
+	return true
+end handler
+
+private foreign handler ObjC_AVCaptureDeviceRequestAccessForMediaType(in pMediaType as ObjcId, in pCompletionHandler as ObjcBlockPointer) \
+		returns nothing \
+		binds to "objc:AVCaptureDevice.+requestAccessForMediaType:completionHandler:"
+
+public handler AudioLibraryRequestPermisions()
+	unsafe
+		ObjC_AVCaptureDeviceRequestAccessForMediaType(StringToNSString("soun"), sRequestPermissionsCompletionHandler)
+	end unsafe
+end handler
+
+public handler RequestPermissionsCompletionHandler(in pBlock as ObjcBlockPointer, in pGranted as CBool)
+	post "permissionsCallback" to sTarget with [pGranted]
+end handler
+
+Parameters:
+pHandler: The handler the block pointer should wrap.
+
+rBlockPtr: The variable into which the block pointer should be returned.
+
+Returns: True if the block pointer was successfully created, false otherwise.
+
+Description:
+Use the <CreateObjcBlockPointerFromHandler> handler to create a pointer to an
+Objective-C block that wraps an LCB handler. The block pointer can be used in
+calls to Objective-C foreign functions that expect a block as a parameter.
+
+The wrapped handler will be called whenever the block is invoked, with the first
+parameter of its signature being the block pointer. The remaining parameters
+should match those of the Objective-C block.
+
+References:
```

Comment:
I suggest making it clear that the ObjcBlockPointer must be deleted when no longer used. e.g.
```
The lifetime of a created `ObjcBlockPointer` is not automatically managed. When such a value
has no more references to it and it is no longer going to be used, `DeleteObjcBlockPointer`
should be used to free the resources used by it.
```